apps/src/music/player/soundSub.js

```diff
@@ -108,23 +112,38 @@ WebAudio.prototype.PlaySoundByBuffer = function (
   when,
   loop,
   effects,
-  callback
+  callback,
+  duration
 ) {
-  var source = audioContext.createBufferSource(); // creates a sound source
+  const source = audioContext.createBufferSource(); // creates a sound source
   source.buffer = audioBuffer; // tell the source which sound to play
+  let currentNode = source;
+
+  if (duration) {
+    // If playing for a specific duration, apply a small fadeout to the sound
+    // to prevent clicks and pops
+    const gainNode = audioContext.createGain();
+    gainNode.gain.setTargetAtTime(
+      0,
+      when + duration - RELEASE_DURATION_SECONDS,
+      RELEASE_TIME_CONSTANT
+    );
```

Comment:
Very cool functionality!

Just learning about this now.  Are there any interesting edge cases where we wouldn't get the full curve from full volume to 0, say if the `duration` is shorter than `RELEASE_DURATION_SECONDS `?

Reply:
Good question. For 120 BPM, this should be fine as 16th notes will be .125 seconds, but at higher tempos, a note duration could definitely be below the RELEASE_DURATION_SECONDS. This should probably be scaled based off of tempo.

Reply:
I created a [task](https://codedotorg.atlassian.net/browse/SL-900) for supporting variable BPM, so I can tweak this as a part of that work.